libfwupdplugin/fu-dummy-efivars.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#define G_LOG_DOMAIN "FuDummyEfivars"
+
+#include "config.h"
+
+#include "fu-dummy-efivars.h"
+
+typedef struct {
+	gchar *guid;
+	gchar *name;
+	guint32 attr;
+	GByteArray *buf;
+} FuDummyEfivarsKey;
+
+struct _FuDummyEfivars {
+	FuEfivars parent_instance;
+	GPtrArray *keys; /* of FuDummyEfivarsKey */
+};
+
+G_DEFINE_TYPE(FuDummyEfivars, fu_dummy_efivars, FU_TYPE_EFIVARS)
+
+static void
+fu_dummy_efivars_key_free(FuDummyEfivarsKey *key)
+{
+	g_free(key->guid);
+	g_free(key->name);
+	g_byte_array_unref(key->buf);
+	g_free(key);
+}
+
+static gboolean
+fu_dummy_efivars_supported(FuEfivars *efivars, GError **error)
+{
+	return TRUE;
+}
+
+static FuDummyEfivarsKey *
+fu_dummy_efivars_find_by_guid_name(FuDummyEfivars *self, const gchar *guid, const gchar *name)
+{
+	for (guint i = 0; i < self->keys->len; i++) {
+		FuDummyEfivarsKey *key = g_ptr_array_index(self->keys, i);
+		if (g_strcmp0(guid, key->guid) == 0 && g_strcmp0(name, key->name) == 0)
+			return key;
+	}
+	return NULL;
+}
+
+static gboolean
+fu_dummy_efivars_delete(FuEfivars *efivars, const gchar *guid, const gchar *name, GError **error)
+{
+	FuDummyEfivars *self = FU_DUMMY_EFIVARS(efivars);
+	FuDummyEfivarsKey *key = fu_dummy_efivars_find_by_guid_name(self, guid, name);
+	if (key != NULL)
```

Comment:
I'd argue this should be setting an error if `key` was `NULL`

Reply:
the linux version of this does:

	if (!g_file_query_exists(file, NULL))
		return TRUE;

i.e. silently ignoring the failure if it's missing. Do you think change that too or preserve the previous behaviour?

Reply:
It's 2.0.0, we can change things. I'd say change it.

Reply:
Done!